### PR TITLE
Add method removeByPath() and support keys as arrays in method remove()

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    |  ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #3

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: src/ArrayHelper.php [hunk 1]
```diff
@@ -455,22 +455,58 @@ private static function parsePath($path, string $delimiter)
...
      * @param array $array the array to extract value from
-     * @param string $key key name of the array element
+     * @param string|array $key key name of the array element or associative array at the key path specified
      * @param mixed $default the default value to be returned if the specified key does not exist
      * @return mixed the value of the element if found, default value otherwise
      */
-    public static function remove(array &$array, string $key, $default = null)
+    public static function remove(array &$array, $key, $default = null)
     {
+        $keys = is_string($key) ? [$key] : $key;
```
Comment (Jokerpwnz) [inline src/ArrayHelper.php:464]: При передачи другого типа (int,float) возможно ошибка  , не ?

Comment (rustamwin) [reply src/ArrayHelper.php:464]: No, it will not.

Comment (vjik) [reply src/ArrayHelper.php]: Thank you. Fixed.

File: src/ArrayHelper.php [hunk 2]
```diff
@@ -455,22 +455,58 @@ private static function parsePath($path, string $delimiter)
...
-     * @param string $key key name of the array element
+     * @param string|int|float|array $key key name of the array element or associative array at the key path specified
      * @param mixed $default the default value to be returned if the specified key does not exist
      * @return mixed the value of the element if found, default value otherwise
      */
-    public static function remove(array &$array, string $key, $default = null)
+    public static function remove(array &$array, $key, $default = null)
     {
-        if (array_key_exists($key, $array)) {
+        $keys = is_array($key) ? $key : [$key];
```
Comment (rustamwin) [inline src/ArrayHelper.php:464]: Support for `Closure` can also be added.

Comment (vjik) [reply src/ArrayHelper.php:464]: `Closure` should removed item and return values... Has use cases?

**Comment by samdark** [comment]
👍

**Pull request merged**